### PR TITLE
[INLONG-8098][Manager] Support Inlong user permission control

### LINKED ISSUE #8098: [Feature][Manager] Support Inlong user permission control
### Description

Different from nomal tenant user, inlong user has the global permission to operate all resources from tenents.

Parent #7914 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8098 
- Parent #7914 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-dao/src/main/resources/mappers/UserRoleEntityMapper.xml
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/UserRoleEntityMapper.xml:1]: Rename UserRoleEntityMapper.xml to TenantUserRoleEntityMapper.xml.

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/UserRoleEntity.java
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/UserRoleEntity.java:1]: Rename UserRoleEntity to TenantUserRoleEntity.

File: inlong-manager/manager-web/sql/changes-1.8.0.sql [hunk 1]
```diff
@@ -44,4 +44,42 @@ CREATE TABLE IF NOT EXISTS `inlong_tenant`
...
+    UNIQUE KEY `unique_tenant_user_role` (`user_name`, `role_code`, `is_deleted`)
+    ) ENGINE = InnoDB
+    DEFAULT CHARSET = utf8mb4 COMMENT ='Inlong User Role Table';
+
+INSERT INTO `inlong_user_role` (`user_name`, `role_code`, `creator`)
+VALUES ('admin', 'INLONG_ADMIN', 'inlong_init');
+
+RENAME TABLE user_role TO tenant_user_role;
+ALTER TABLE tenant_user_role
+    ADD tenant VARCHAR(256) DEFAULT 'public' NOT NULL comment 'User tenant';
```
Comment (fuweng11) [inline inlong-manager/manager-web/sql/changes-1.8.0.sql:72]: `comment` to  `COMMENT`

File: inlong-manager/manager-dao/src/main/resources/mappers/UserRoleEntityMapper.xml
Comment (healchow) [reply inlong-manager/manager-dao/src/main/resources/mappers/UserRoleEntityMapper.xml:1]: Is the table name `user_role` or `tenant_user_role`?

The name of the Entity should be consistent with the name of the table.

Comment (healchow) [reply inlong-manager/manager-dao/src/main/resources/mappers/UserRoleEntityMapper.xml:1]: I see that the table name is `tenant_user_role`, so your comment is right.

File: inlong-manager/manager-web/sql/changes-1.8.0.sql (hunk 1 above)
Comment (vernedeng) [reply inlong-manager/manager-web/sql/changes-1.8.0.sql]: fixed, thx

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongUserRoleEntityMapper.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.github.pagehelper.Page;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface InlongUserRoleEntityMapper {
+
+    int insert(InlongUserRoleEntity record);
+
+    InlongUserRoleEntity selectById(Integer id);
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongUserRoleEntityMapper.java:31]: Whether a more username lookup is required？

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongUserRoleEntityMapper.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    int updateById(InlongUserRoleEntity record);
+
+    Page<InlongUserRoleEntity> listByCondition(InlongRolePageRequest request);
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongUserRoleEntityMapper.java:35]: fixed, thx

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+@RequestMapping("/api")
+@Api(tags = "INLONG-USER-API")
+public class InlongRoleController {
+
+    @Autowired
+    private InlongRoleService inlongRoleService;
+
+    @RequestMapping(value = "/role/inlong/get/{id}", method = RequestMethod.GET)
+    @ApiOperation(value = "Get tenant role")
+    @ApiImplicitParam(name = "id", dataTypeClass = Integer.class, required = true)
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java:50]: fixed, thx

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql [hunk 5]
```diff
@@ -580,6 +580,29 @@ CREATE TABLE IF NOT EXISTS `user_role`
...
+    `role_code`   varchar(256) NOT NULL COMMENT 'User role code',
+    `disabled`    tinyint(1)   NOT NULL DEFAULT '0' COMMENT 'Whether to disabled, 0: enabled, 1: disabled',
+    `is_deleted`  int(11)               DEFAULT '0' COMMENT 'Whether to delete, 0 is not deleted, if greater than 0, delete',
+    `creator`     varchar(256) NOT NULL COMMENT 'Creator name',
+    `modifier`    varchar(256)          DEFAULT NULL COMMENT 'Modifier name',
+    `create_time` datetime     NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'Create time',
+    `modify_time` datetime     NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'Modify time',
+    `version`     int(11)      NOT NULL DEFAULT '1' COMMENT 'Version number, which will be incremented by 1 after modification',
+    PRIMARY KEY (`id`),
+    UNIQUE KEY `unique_user_role` (`user_name`, `role_code`, `is_deleted`)
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-manager/manager-web/sql/apache_inlong_manager.sql:599]: fixed, thx

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongUserRoleEntityMapper.java (hunk 2 above)
Comment (vernedeng) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongUserRoleEntityMapper.java:31]: At least not now, there are only two roles in inlong user role managerment.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    public PageInfo<InlongRoleInfo> listByCondition(InlongRolePageRequest request) {
+        PageHelper.startPage(request.getPageNum(), request.getPageSize());
+        Page<InlongUserRoleEntity> entityPage = inlongUserMapper.selectByCondition(request);
+        return entityPage.toPageInfo(entity -> CommonBeanUtils.copyProperties(entity, InlongRoleInfo::new));
+    }
+
+    @Override
+    public int save(InlongRoleRequest request) {
+        String username = request.getUsername();
+        Preconditions.expectNotBlank(username, "Failed to save tenant user role, user should not be blank");
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleServiceImpl.java:53]: "Failed to save inlong user role, user should not be blank"

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        Preconditions.expectNotBlank(request.getRoleCode(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleServiceImpl.java:54]: "Failed to save inlong user role, role code should not be blank"

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+@RestController
+@RequestMapping("/api")
+@Api(tags = "INLONG-USER-API")
+public class InlongRoleController {
+
+    @Autowired
+    private InlongRoleService inlongRoleService;
+
+    @RequestMapping(value = "/role/inlong/get/{id}", method = RequestMethod.GET)
+    @ApiOperation(value = "Get tenant role")
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java:51]: "Get inlong role"

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    @RequiresRoles(value = UserRoleCode.INLONG_ADMIN)
+    @ApiImplicitParam(name = "id", dataTypeClass = Integer.class, required = true)
+    public Response<InlongRoleInfo> get(@PathVariable int id) {
+        return Response.success(inlongRoleService.get(id));
+    }
+
+    @RequestMapping(value = "/role/inlong/save", method = RequestMethod.POST)
+    @OperationLog(operation = OperationType.CREATE)
+    @RequiresRoles(value = UserRoleCode.INLONG_ADMIN)
+    @ApiOperation(value = "Save tenant role")
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java:61]: "Save inlong user role"

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    public Response<Integer> save(@Validated @RequestBody InlongRoleRequest request) {
+        return Response.success(inlongRoleService.save(request));
+    }
+
+    @RequestMapping(value = "/role/inlong/list", method = RequestMethod.POST)
+    @ApiOperation(value = "List tenant by paginating")
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongRoleController.java:67]: "List inlong user role info by paginating"

**Comment by dockerzhang** [comment]
Please rebase from the master to fix the failed UT after https://github.com/apache/inlong/pull/8128 is merged.

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/InlongRoleRequest.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    @NotBlank
+    private String roleCode;
+
+    @ApiModelProperty(value = "If disabled")
+    private Integer disabled = 0;
+
+    @ApiModelProperty(value = "Version number")
+    private Integer version;
+
+    public InlongRoleRequest gentRequest() {
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/InlongRoleRequest.java:53]: `gentRequest` -> `genRequest`

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/UserRoleCode.java [hunk 12]
```diff
@@ -25,4 +25,8 @@ public class UserRoleCode {
     public static final String ADMIN = "ADMIN";
     public static final String OPERATOR = "OPERATE";
 
+    public static final String INLONG_ADMIN = "INLONG_ADMIN";
+
+    public static final String INLONG_OPERATOR = "INLONG_OPERATE";
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/UserRoleCode.java:30]: What is the difference between `ADMIN` and `INLONG_ADMIN`?

What's more, the `"OPERATOR"` is more clear than the `"OPERATE"`.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/InlongRoleRequest.java (hunk 11 above)
Comment (vernedeng) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/InlongRoleRequest.java:53]: fixed, thx

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/UserRoleCode.java (hunk 12 above)
Comment (vernedeng) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/user/UserRoleCode.java:30]: **_ADMIN_** only has the permission of its own tenant, while **_INLONG_ADMIN_** has permission of all tenant. 
The permission check will be done at those **servlet filters**

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**